state.tf

```diff
@@ -1,3 +1,9 @@
 terraform {
   backend "azurerm" {}
 }
+
+data "azurerm_subnet" "main_subnet" {
```

Comment:
IIRC this is the subnet for the ASE? Can we name it as such as main_subnet is a bit confusing.